MAVProxy/modules/mavproxy_arm.py

```diff
@@ -120,6 +120,25 @@ def cmd_disarm(self, args):
             0, # param6
             0) # param7
 
+    def all_checks_enabled(self):
+        ''' returns true if the UAV is skipping any arming checks'''
+        arming_mask = int(self.get_mav_param("ARMING_CHECK",0))
+        if arming_mask == 1:
+            return True
+        for bit in arming_masks.values():
+            if not arming_mask & bit and bit != 1:
+                return False
+        return True
+
+    def mavlink_packet(self, m):
+        mtype = m.get_type()
+        if mtype == 'HEARTBEAT' and m.type != mavutil.mavlink.MAV_TYPE_GCS:
```

Comment:
On Wed, 7 Sep 2016, Francisco Ferreira wrote:

> Is this correct? What if the heartbeat is an antenna tracker or gimbal type
> or etc.?

Wildly incorrect.  But I swiped it from mavproxy_link.py, so at least 
we're consistently incorrect.

I think the correct fix might be to check the hearbeat actually came from 
*our* mav.  I would dearly love to find lots of time to use the mav source 
system pymavlink code 
(https://github.com/ArduPilot/mavlink/compare/master...peterbarker:mavsystem-object?expand=1) 
to give MAVProxy proper multi-vehicle support.

We're relying on master.motors.armed() to be correct here, so I think the 
net result of that check being bodgy is that we might run that check more 
often than we need to.